pulsar-function-go/pf/instance.go

```diff
@@ -312,6 +312,24 @@ func (gi *goInstance) setupConsumer() (chan pulsar.ConsumerMessage, error) {
 		subscriptionName = funcDetails.Source.SubscriptionName
 	}
 
+	subscriptionPosition := pulsar.SubscriptionPositionLatest
+	if gi.context.instanceConf.funcDetails.Source.SubscriptionPosition == pb.SubscriptionPosition_EARLIEST {
+		subscriptionPosition = pulsar.SubscriptionPositionEarliest
+	}
+
+	var dlqPolicy *pulsar.DLQPolicy
+	retryEnable := false
+	if retryDetails := gi.context.instanceConf.funcDetails.RetryDetails; retryDetails != nil {
+		dlqPolicy = &pulsar.DLQPolicy{
+			DeadLetterTopic: retryDetails.DeadLetterTopic,
+		}
+
+		if retryDetails.MaxMessageRetries >= 0 {
+			dlqPolicy.MaxDeliveries = uint32(retryDetails.MaxMessageRetries)
+		}
+		retryEnable = true
```

Comment:
isn't `RetryEnable` related to having RetryLetterTopic defined?

https://github.com/apache/pulsar-client-go/blob/393f80b4b93faa36936380b643426026a2b2cd02/pulsar/retry_router.go#L62-L75

I'd assume that `DeadLetterTopic` can be set without setting `RetryEnable`.

Reply:
oh, it doesn't seem to be the case in pulsar-client-go: https://github.com/apache/pulsar-client-go/blob/393f80b4b93faa36936380b643426026a2b2cd02/pulsar/consumer_impl.go#L116-L168

In the Java client, this is different. It seems to make sense that DLQ feature is independent of retrying, but that's not how pulsar-client-go is implemented.

Reply:
seems it's more reasonable to leave `RetryEnable` to false here, then it just won't set the default `DeadLetterTopic`